### PR TITLE
Use correct current page state

### LINKED ISSUE #5216: Navigation: Remove selected state from pages that are not the home page
**Is your feature request related to a problem? Please describe.**
Currently, the first item in the navigation is highlighted as active no matter which page the user is on. This is incorrect for all pages other than the homepage. After discussions with the UX team and various stakeholders, it has been decided that we can't consistently determine which navigation item a particular page falls under. To remove the confusing scenario where the first item is selected when the user is not on that page, we will only highlight the homepage link in the nav if the user is on that page, otherwise we will not highlight anything.

This is a solution that we will be adopting temporarily until we are able to define a solution to know which navigation item should be highlighted as active.

**Describe the solution you'd like**
Check to see if the visited URL matches the URL for the homepage in config, and mark that nav item as active if the user is on that page. Otherwise, do not mark any items as active.

**Describe alternatives you've considered**

**Testing notes**
Test that the homepage link is marked as active when visiting the homepage, and nothing is marked as active for other pages.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**

### LINKED ISSUE #5660: Storybook's "Pages|Front Page" does not select the correct state for navigation
**Describe the bug**
Storybook's "Pages|Front Page" does not select the correct state for navigation. When on the home page, the first navigation item should always be selected, however the storybook request isn't the correct url to match on so nothing is selected.

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to http://localhost:9001/?path=/story/pages-front-page--serbian-cyr
2. Notice there is no selected state on the navigation

**Expected behaviour**
The first link should be selected.

**Screenshots**
Current:
image

Expected:
image

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by AlistairGempf**
Resolves #5216 and resolves #5660

**Overall change:** 
Highlights the correct navigation link as the current page when the url matches that in config exactly.

**Code changes:**
- Checks the RequestConfig canonical url against the config navigation urls and finds the index of a match

_Note: This may not work properly if the navigation url is actually a redirect to another page._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

On the homepage:
image

On another page that is in the nav config:
image (1)

On any other page:
image (2)

On a page that starts with a url that starts with something in the nav config:
image

### CODE REVIEW

**Comment by AlistairGempf** [comment]
~One unintentional consequence of this is that the storybook example of the front page no longer shows the selected state for the initial page. I don't really think there's a good way around this though~

~Never mind, there was a simple solution! Just provide the first navigation url to the `RequestContext` in the story. <- This doesn't seem to work.~

~I've tried loading the config and passing the navigation url as `pathname` to `RequestContext`, but it doesn't seem to actually read it.~

It needed to be in the `FrontPage` props - https://github.com/bbc/simorgh/pull/5634/files#diff-ab45cb8648398012be4f9c0de888d2b7R85.

**Review by DenisHdz — APPROVED** [review]
LGTM

File: src/app/containers/Navigation/index.jsx [hunk 1]
```diff
@@ -32,12 +40,18 @@ const NavigationContainer = () => {
...
+  const { canonicalLink, origin } = useContext(RequestContext);
   const { currentPage, sections } = translations;
 
   if (!navigation || navigation.length === 0) {
     return null;
   }
 
+  const matchingLink = linkUrl => `${origin}${linkUrl.url}` === canonicalLink;
+  const activeIndex = navigation
+    .map(matchingLink)
```
Comment (tochwill) [inline src/app/containers/Navigation/index.jsx]: Do you need to map first here or could you just have the logic within findIndex?

File: src/app/containers/Navigation/index.jsx [hunk 2]
```diff
@@ -32,12 +40,17 @@ const NavigationContainer = () => {
...
   );
+  const { canonicalLink, origin } = useContext(RequestContext);
   const { currentPage, sections } = translations;
 
   if (!navigation || navigation.length === 0) {
     return null;
   }
 
+  const activeIndex = navigation.findIndex(
+    linkUrl => `${origin}${linkUrl.url}` === canonicalLink,
```
Comment (tochwill) [inline src/app/containers/Navigation/index.jsx]: Super minor, could you call linkUrl 'link' instead?

**Review by tochwill — APPROVED** [review]

**Pull request merged**